components/brave_wallet/browser/polkadot/polkadot_substrate_rpc.cc

```diff
@@ -327,6 +327,59 @@ void PolkadotSubstrateRpc::OnGetAccountBalance(
   return std::move(callback).Run(nullptr, WalletParsingErrorMessage());
 }
 
+void PolkadotSubstrateRpc::GetFinalizedHead(std::string_view chain_id,
+                                            GetFinalizedHeadCallback callback) {
+  auto url = GetNetworkURL(chain_id);
+
+  auto payload = base::WriteJson(
+      MakeRpcRequestJson("chain_getFinalizedHead", base::ListValue()));
+  CHECK(payload);
+
+  api_request_helper_.Request(
+      net::HttpRequestHeaders::kPostMethod, url, *payload, "application/json",
+      base::BindOnce(&PolkadotSubstrateRpc::OnGetFinalizedHead,
+                     weak_ptr_factory_.GetWeakPtr(), std::move(callback)));
+}
+
+void PolkadotSubstrateRpc::OnGetFinalizedHead(GetFinalizedHeadCallback callback,
+                                              APIRequestResult api_result) {
+  if (!api_result.Is2XXResponseCode()) {
+    return std::move(callback).Run(std::nullopt, WalletInternalErrorMessage());
+  }
+
+  auto res = polkadot_substrate_rpc_responses::PolkadotFinalizedHead::FromValue(
+      api_result.value_body());
+
+  if (!res) {
+    return std::move(callback).Run(std::nullopt, WalletParsingErrorMessage());
+  }
+
+  if (res->error) {
+    if (res->error->message) {
+      return std::move(callback).Run(std::nullopt, res->error->message.value());
+    }
+    return std::move(callback).Run(std::nullopt, WalletInternalErrorMessage());
+  }
+
+  if (!res->result) {
+    return std::move(callback).Run(std::nullopt, WalletParsingErrorMessage());
+  }
+
```

Comment:
This would be copy-pasted a lot. Should have some some anonymous templated handler.
Ok if next time.

Reply:
Gonna mark this one as resolved for now. We'll likely come up with better and better abstractions as time goes on and we start using the RPC calls for real.